### DISCUSSION TITLE
Sprite issue with artefacts

### DISCUSSION THREAD

**Discussion by videobuff** [Bodmer/TFT_eSPI#1677] [General]
I have build a dashboard with meters, using the sprite principle. I used the exaple "animated dial" as guide.  They really work nice.
Now i have added a long touch touch sense on the touch screen which brings me to a menu. The menu is another jpeg image with three buttons drawn on it, which i load, and i use x/y sensing for the choices. These are a keyboard for entry, a memory screen to choose presets and  going back to the dashboard with the meters. If i choose the latter i get graphical artefacts in the sprite area but once the meters go over that artefact area, all is fine.

I have made a demo with the example animated dial sprite , then switching to a menu for a few seconds, and then coming back to the dial. It is then, that sometimes artefacts are seen. How can I solve this.
The sketch is attached. I run it on a WT32-SCO1 display but i had the same a a 3.5" TFT SPI "red" display 

Erik

![Schermafbeelding 2022-03-02 om 08 50 50](https://user-images.githubusercontent.com/3132953/156318126-3dea305f-059e-4334-a220-3a256f96e808.jpg)

[Sprite_Issue.zip](https://github.com/Bodmer/TFT_eSPI/files/8167645/Sprite_Issue.zip)

**Comment by Bodmer** [comment]
The problem is that a new needle is created in every loop() within compas_set.h so a new buffer is created. Eventually the sketch will crash. Only create the needle once in setup().

**Reply by videobuff** [reply]
Thanks for the advice. Works as you wrote. Now I also realize why I fumbled around. If the ange value is fixed, then when i return from the menu, the needle is not drawn. I have sorted that with a one shot function which plots the needle one time with an angle = angle +1. Hardly noticable, but i will look it up in the procedure and implement that the first time it should always plot the needle, coming from the menu. Thanks again for a wonderfull library and it keeps my 72 year old brain sharp.
Erik

to test: add ` int angle = 90; `to the constants and stop the random angle function. 
`//uint16_t angle = random(241); // random speed in range 0 to 240`

Erik

**Reply by Bodmer** [reply]
OK, I understand now. A good approach will be to draw the dial jpeg, then use:
```
buffer_loaded = false;
```
Then draw the needle.

This works because buffer_loaded = false forces the needle to be drawn.

When the sketch is first run it is set false here:
[buffer_loaded = false;](https://github.com/Bodmer/TFT_eSPI/blob/master/examples/Sprite/Animated_dial/Animated_dial.ino#L45)

Being false means the old screen buffer will not be pasted back (potentially that would be pasted back in the wrong position if the needle is moved) before drawing the needle.
